### PR TITLE
Feature/blui 4996 use app theme

### PR DESCRIPTION
**Pull request by ektaghag-eaton**
Fixes 4996 .

#### Changes proposed in this Pull Request:
- Use ExtendedTheme hook
- Remove ts-ignore comments
- 

#### Screenshots / Screen Recording (if applicable)
-

#### To Test:
- use showcase PR https://github.com/etn-ccis/blui-react-native-showcase-demo/pull/148
- All components using proper colors

#### Any specific feedback you are looking for?
-

### CODE REVIEW

**Comment by huayunh** [comment]
(this is too technical for me; please let a developer to review this)

**Comment by ArshdeepSingh3-eaton** [comment]
tested out 
- ChannelValue
- Chip
- Collapsible Header layout
- Drawer
- EmptyState
- Grades

and are working as they should

**Comment by ektaghag-eaton** [comment]
Hi @surajeaton , @manojleaton and @ArshdeepSingh3-eaton 
I have checked the following components:
- ListItemTag
- MobileStepper
- Overline
- ScoreCard
- UserMenu

And found a few jsDoc changes that I have Updated in this PR. A few issues are logged in github.

**Review by ArshdeepSingh3-eaton — APPROVED** [review]

**Comment by surajeaton** [comment]
> Hi @surajeaton , @manojleaton and @ArshdeepSingh3-eaton I have checked the following components:
> 
> * ListItemTag
> * MobileStepper
> * Overline
> * ScoreCard
> * UserMenu
> 
> And found a few jsDoc changes that I have Updated in this PR. A few issues are logged in github. #471 [etn-ccis/blui-react-native-showcase-demo#150](https://github.com/etn-ccis/blui-react-native-showcase-demo/issues/150)

Checked:
- Hero
- Hero Banner
- Icon
- InfoListItem

Updated the issue https://github.com/etn-ccis/blui-react-native-component-library/issues/471 with the findings.

**Comment by manojleaton** [comment]
Checked the `Header` component and created a issue [#480](https://github.com/etn-ccis/blui-react-native-component-library/issues/480) with the findings. Thanks

File: components/src/core/ListItemTag/ListItemTag.tsx [hunk 1]
```diff
@@ -68,11 +68,8 @@ const listItemTagStyles = (
             color:
                 props.fontColor ||
                 (Color(props.backgroundColor || theme.colors.primary).isLight()
-                    ? // @TODO Currently neutral30 is #414E54 and as per color pallete black[500] is #424E54 Add Figma Variable
-                      // @ts-ignore
-                      BLUIColors.neutral[30]
-                    : // @ts-ignore
-                      BLUIColors.primary[100]),
+                    ? theme.colors.onNeutralShadedContainer
```
Comment (ektaghag-eaton) [inline components/src/core/ListItemTag/ListItemTag.tsx:71]: @huayunh , can you check these color codes are correct? If not please provide me a Figma link of the default color. when we worked on this I didn't have a color code and we just chose close color code if you see removed line 71.

File: components/src/core/InfoListItem/InfoListItem.tsx [hunk 2]
```diff
@@ -339,13 +339,9 @@ export const InfoListItem: React.FC<InfoListItemProps> = (props) => {
...
             return statusColor
                 ? color(statusColor).isDark()
-                    ? // @ts-ignore TODO
-                      BLUIColors.primary[100]
-                    : // @TODO Currently neutral30 is #4242E4 and as per color pallete black[500] is #4142E4 Add Figma Variable
-                      // @ts-ignore
-                      BLUIColors.neutral[30]
-                : // @ts-ignore TODO
-                  BLUIColors.primary[100]; // default avatar is dark gray -> white text
+                    ? theme.colors.onPrimaryFilledContainer
```
Comment (ektaghag-eaton) [inline components/src/core/InfoListItem/InfoListItem.tsx:342]: Hi @huayunh can you check if these color codes are correct? If not please provide me a Figma link of the default color. when we worked on this I didn't have a color code and we just chose close color code if you see removed line 342.

**Comment by ektaghag-eaton** [comment]
For fixing https://github.com/etn-ccis/blui-react-native-component-library/pull/469#issuecomment-1866162610 we have another stories in backlog

**Pull request merged**